### PR TITLE
Added userstats to mysql input plugin

### PR DESCRIPTION
**Pull request by acezellponce**
### Required for all PRs:
- [ ] CHANGELOG.md updated (we recommend not updating this until the PR has been approved by a maintainer)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] README.md updated (if adding a new plugin)

### CODE REVIEW

**Comment by nothing2see** [comment]
What are the chances of getting this merged in with the next release? Would love to make use of these metrics without having to custom build.

**Comment by nhaugo** [comment]
@acezellponce I would like to get this scheduled for a release but it would be nice if it included a couple of tests.

**Comment by acezellponce** [comment]
@nhaugo I would be able to do it but it will take some time since I am new to Go programming language.

**Comment by acezellponce** [comment]
I was going to try writing some tests, but I'm not really sure what to add?

**Comment by sparrc** [comment]
I'm not sure that there is a simple way to add unit tests here without mocking up a mysql db.

File: plugins/inputs/mysql/mysql.go [hunk 1]
```diff
@@ -908,6 +932,77 @@ func (m *Mysql) gatherGlobalStatuses(db *sql.DB, serv string, acc telegraf.Accum
...
+				return err
+			}
+
+			tags := map[string]string{"server": servtag, "user": user}
+			fields := make(map[string]interface{})
+
+			if err != nil {
+				return err
+			}
+			fields["total_connections"] = total_connections
```
Comment (sparrc) [inline plugins/inputs/mysql/mysql.go]: here you should inline create the `fields` map with all values, like this:

```go
fields := map[string]interface{}{
  "concurrent_connections": concurrent_connections,
  // etc
}
```

File: plugins/inputs/mysql/mysql.go [hunk 2]
```diff
@@ -908,6 +932,77 @@ func (m *Mysql) gatherGlobalStatuses(db *sql.DB, serv string, acc telegraf.Accum
...
+			)
+
+			if err != nil {
+				return err
+			}
+
+			tags := map[string]string{"server": servtag, "user": user}
+			fields := make(map[string]interface{})
+
+			if err != nil {
```
Comment (sparrc) [inline plugins/inputs/mysql/mysql.go]: you're checking the same error twice here

File: plugins/inputs/mysql/mysql.go [hunk 3]
```diff
@@ -908,6 +932,77 @@ func (m *Mysql) gatherGlobalStatuses(db *sql.DB, serv string, acc telegraf.Accum
 		}
 	}
 
+	// gather connection metrics from user_statistics for each user
+	if m.GatherUserStatistics {
```
Comment (sparrc) [inline plugins/inputs/mysql/mysql.go]: aren't you doing the same thing twice here? above you check the same boolean and call it out to a separate function?

File: plugins/inputs/mysql/mysql.go [hunk 4]
```diff
@@ -690,6 +704,10 @@ func (m *Mysql) gatherGlobalVariables(db *sql.DB, serv string, acc telegraf.Accu
 		}
 		key = strings.ToLower(key)
 		// parse value, if it is numeric then save, otherwise ignore
+		if strings.Contains(key, "version") {
```
Comment (sparrc) [inline plugins/inputs/mysql/mysql.go]: what is this? why did you add this?

**Comment by sparrc** [comment]
closing for lack of updates

File: plugins/inputs/mysql/mysql.go (hunk 2 above)
Comment (acezellponce) [inline plugins/inputs/mysql/mysql.go]: Okay, I've removed it.

File: plugins/inputs/mysql/mysql.go (hunk 1 above)
Comment (acezellponce) [inline plugins/inputs/mysql/mysql.go]: Made that changed, thanks for this suggestion.

**Comment by acezellponce** [comment]
@sparrc Sorry, I was out of the country for a while but back now and working on this again.

**Comment by sparrc** [comment]
please open a new pr
